ngi_reports/common/__init__.py

```diff
@@ -58,18 +79,16 @@ def parse_piper_xml(self):
                 try:
                     project['id'] = run['metadata']['name']
                     project['ngi_name'] = run['metadata']['name']
-                    # One sample
-                    if isinstance(run['inputs']['sample'], dict):
-                        sid = run['inputs']['sample']['samplename']
-                        samples[sid] = {'id': sid}
-                    # Many samples
-                    elif isinstance(run['inputs']['sample'], list):
-                        for sample in run['inputs']['sample']:
-                            sid = sample['samplename']
-                            samples[sid] = {'id': sid}
-                    else:
-                        # This is passed below so doesn't halt execution
-                        raise KeyError("Could not parse run['inputs']['sample']")
+                    for sample in _nextitem(run['inputs']['sample']):
+                        try:
+                            fcid = [_getfcid(pfu['unitinfo']) \
+                                for library in _nextitem(sample['library']) \
+                                for pfu in _nextitem(library['platformunit'])]
+                        except (IndexError, KeyError) as e:
+                            self.LOG.warning('Could not parse platform unit info in sample XML file: {}'.format(e.message))
+                            fcid = ['N/A']
+                        sid = sample['samplename']
+                        samples[sid] = {'id': sid, 'run_id': ';'.join(set(fcid))}
```

Comment:
What's the reason for logging `'id': sid` as well as using `sid` in the dict key? Just convienience when reading the dict? eg. Why not use `samples[sid] = fcid`?